torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1,15 +1,64 @@
+#include <cstdbool>
 #include <torch/csrc/jit/passes/quantization.h>
 
 #include <torch/csrc/jit/ir.h>
 #include <torch/csrc/jit/node_hashing.h>
 #include <torch/csrc/jit/passes/alias_analysis.h>
 
 #include <unordered_map>
+#include <unordered_set>
 
 namespace torch {
 namespace jit {
-namespace {} // namespace
+namespace {
+// QuantizerUtils
 
+int checkIfNodeValuesQuantizable(torch::jit::Node* n) {
+  // This is lookup for quantizable nodes. It will be
+  // expanded in future to support more ops and patterns
+  static const std::unordered_set<std::string> nodeLookup =
+    {"aten::_convolution", "aten::relu"};
+  AT_ASSERT(n != nullptr);
+  return nodeLookup.count(n->schema().name());
+}
+
+// Add Quant-Dequant node pair for quantizable node
+void addQuantDeQuantNodes(torch::jit::Value* v, torch::jit::Node* n=nullptr) {
+  bool beforeValue = false;
+  if (n == nullptr) {
+    // If caller supplies, then insertion point is v : n else we assume its
+    // between n{v->node()} : v
+    n = v->node();
+    beforeValue = true;
+  }
+  torch::jit::Node* quant =
+      n->owningGraph()->create(at::Symbol::fromQualString("aten::quant"));
+  AT_ASSERT(quant != nullptr);
+  torch::jit::Node* dequant =
+      n->owningGraph()->create(at::Symbol::fromQualString("aten::dequant"));
+  AT_ASSERT(dequant != nullptr);
+  n->owningBlock()->appendNode(quant);
+  n->owningBlock()->appendNode(dequant);
+  quant->setScope(n->scope());
+  dequant->setScope(n->scope());
+  if (beforeValue) {
+    quant->moveAfter(n);
+    dequant->moveAfter(quant);
+    v->replaceAllUsesWith(dequant->outputs()[0]);
+  } else {
+    dequant->moveBefore(n);
+    quant->moveBefore(dequant);
+    n->replaceInputWith(v, dequant->outputs()[0]);
+  }
+  quant->outputs()[0]->setUniqueName(v->uniqueName() + ".quant");
```

Comment:
Where is  the output added to the quant node?

Reply:
as part of create(). Default number of outputs = 1